compiler/cpp/CMakeLists.txt

```diff
@@ -16,30 +16,30 @@
 # specific language governing permissions and limitations
 # under the License.
 #
+cmake_minimum_required(VERSION 2.8.12)
+
+set(BIN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/bin)
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/thrift/version.h.in ${CMAKE_CURRENT_BINARY_DIR}/thrift/version.h)
 
 find_package(FLEX REQUIRED)
 find_package(BISON REQUIRED)
 
 # Create flex and bison files and build the lib parse static library
-BISON_TARGET(thrifty ${CMAKE_CURRENT_SOURCE_DIR}/src/thrift/thrifty.yy ${CMAKE_CURRENT_BINARY_DIR}/thrift/thrifty.cc)
+BISON_TARGET(thrifty
+        ${CMAKE_CURRENT_SOURCE_DIR}/src/thrift/thrifty.yy
+        ${CMAKE_CURRENT_BINARY_DIR}/thrift/thrifty.cc
+        DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/thrift/thrifty.h)
```

Comment:
Nice addition of DEFINES_FILE so rebuilds are not triggered - I was wondering why successive build runs would rebuild the compiler - this is probably one fix for such a case.